open_widget_framework/open_widget_framework/views.py

```diff
@@ -1,204 +1,172 @@
+from open_widget_framework.widget_serializer import WidgetSerializer, WidgetListSerializer, \
+    get_widget_class_configurations
+from open_widget_framework.settings import api_settings
 
 # TODO: validate with widget list
 
 
-def get_widget_lists(request):
+def get_widget_configurations(request):
     """
-    API endpoint for returning a list of all WidgetList ids
+    API endpoint for getting all available widget classes and their configurations
     """
-    widget_lists = WidgetList.objects.all().values("id")
-    return JsonResponse(list(widget_lists), safe=False)
+    return JsonResponse({'widgetClassConfigurations': get_widget_class_configurations()})
 
 
-def get_widget_configurations(request):
+def make_widget_list_response(queryset):
     """
-    API endpoint for getting all available widget classes and their configurations
+    make_widget_list_response takes a queryset of widgetInstances and returns a list of widgets serialized and rendered
+        with their title. This is the response for most of the widget level api endpoints so that the frontend can
+        update it's widget-list
     """
-    return JsonResponse(
-        {"widgetClassConfigurations": get_widget_class_configurations()}
-    )
+    return JsonResponse([WidgetSerializer(widget).render_with_title() for widget in queryset], safe=False)
 
 
-class WidgetListView(View):
-    @staticmethod
-    def get(request, widget_list_id):
-        """
-        API endpoint for getting the widgets in a single list
-
-        Args:
-            widget_list_id (int): id of desired WidgetList
-        """
-        widget_list = get_object_or_404(WidgetList, id=widget_list_id)
-        # TODO: widget_list.can_access
-        # Construct a dictionary with id and configuration props to be consumed and rendered by React components
-        widget_instances = [
-            {
-                "id": widget.id,
-                "position": widget.position,
-                "widgetProps": widget.make_render_props(),
-            }
-            for widget in widget_list.get_widgets()
-        ]
+class WidgetListViewSet(ModelViewSet):
+    """
+    WidgetListViewSet handles requests at the widget-list level with the following mapping (as reflected in urls.py):
+        get_lists (GET with no list ID) -> list
+        GET (with list ID) -> retrieve
+        POST -> create
+        DELETE -> destroy
 
-        return JsonResponse(widget_instances, safe=False)
+        It also implements authentication and permissions if they are defined in the widget-framework settings
+    """
+    queryset = WidgetList.objects.all()
+    serializer_class = WidgetListSerializer
+    if api_settings.WIDGET_FRAMEWORK_AUTHENTICATION_CLASSES:
+        authentication_classes = api_settings.WIDGET_FRAMEWORK_AUTHENTICATION_CLASSES
+    if api_settings.WIDGET_FRAMEWORK_PERMISSION_CLASSES:
+        permission_classes = api_settings.WIDGET_FRAMEWORK_PERMISSION_CLASSES
```

Comment:
Suggest you split this into one set of permission classes for `WidgetList` models and one set of permission classes for `WidgetInstance`

Reply:
@rhysyngsun Explain what you mean a bit more. I was under the impressions that you only wanted to be able to control permissions at the widget list level. You don't want to set object level permissions for individual widget do you? So while the appropriate serializer is the widget serializer, we want to be checking permissions on the widget-list which is not the model that the view set is based on. That's why I did it the way that I did.

Reply:
Also, if I'm deleting these since the APIView checks for defaults anyway, what exactly would I be splitting? Wouldn't those just be something you define and set in your default settings?

Reply:
@rhysyngsun Merging in this PR for now, but if this is unresolved, we can address it in a later PR